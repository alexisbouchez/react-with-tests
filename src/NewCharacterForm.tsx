import type { FC } from 'react'
import { useState } from 'react'

interface NewCharacterFormProps {}

const NewCharacterForm: FC<NewCharacterFormProps> = () => {
  const [name, setName] = useState('')

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    console.log(name)
  }

  return (
    <form role="form" onSubmit={onSubmit}>
      <div>
        <label htmlFor="name">Name</label>
        <input type="text" id="name" role="textbox" data-testid="name-input" />
      </div>

      <div>
        <button type="submit" role="button">
          Submit
        </button>
      </div>
    </form>
  )
}

export default NewCharacterForm
