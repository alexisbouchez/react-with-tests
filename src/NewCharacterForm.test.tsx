import { render, screen } from '@testing-library/react'
import NewCharacterForm from './NewCharacterForm'

describe('NewCharacterForm', () => {
  it('renders the form correctly', () => {
    render(<NewCharacterForm />)

    const form = screen.getByRole('form')
    expect(form).toBeInTheDocument()
  })

  it('renders a name field correctly', () => {
    render(<NewCharacterForm />)

    const label = screen.getByLabelText(/name/i)
    expect(label).toBeInTheDocument()

    const input = screen.getByTestId('name-input')
    expect(input).toBeInTheDocument()
    expect(input).toHaveAttribute('type', 'text')
    expect(input).toHaveAttribute('id', 'name')
  })

  it('renders a submit button correctly', () => {
    render(<NewCharacterForm />)

    const button = screen.getByRole('button')
    expect(button).toBeInTheDocument()
  })
})
