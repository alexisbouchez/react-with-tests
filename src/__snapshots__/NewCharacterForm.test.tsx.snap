// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCharacterForm renders the form correctly 1`] = `
<div>
  <form>
    <div>
      <label
        for="name"
      >
        Name
      </label>
      <input
        id="name"
        type="text"
      />
    </div>
    <div>
      <button
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
